src/Tests/Microsoft.DotNet.ApiCompatibility.Tests/Logging/SuppressionEngineTests.cs

```diff
@@ -110,6 +110,36 @@ public void SuppressionEngineSupportsGlobalCompare()
             Assert.False(engine.IsErrorSuppressed("CP0003", "T:A.D", "ref/net6.0/myLeft.dll", "lib/net6.0/myRight.dll", isBaselineSuppression: true));
         }
 
+        [Fact]
+        public void IsErrorSuppressedForLeftAndRightWithoutDiagnosticId()
+        {
+            SuppressionEngine engine = new();
+            string left = "ref/net6.0/A.dll";
+            string right = "lib/net6.0/A.dll";
+
+            // Explicitly add a suppression without a diagnostic id for testing.
+            engine.AddSuppression(string.Empty, left: left, right: right, isBaselineSuppression: true);
+            Assert.True(engine.IsErrorSuppressed("CP9999", left: left, right: right, isBaselineSuppression: true));
+            Assert.False(engine.IsErrorSuppressed("CP9999", left: left, right: right, isBaselineSuppression: false));
+
+            engine.AddSuppression(string.Empty, left: left, right: right, isBaselineSuppression: false);
+            Assert.True(engine.IsErrorSuppressed("CP9999", left: left, right: right, isBaselineSuppression: false));
+        }
+
+        [Fact]
+        public void CtorSuppressionWithoutDiagnosticIdRoundtrip()
+        {
+            SuppressionEngine engine = new();
+            // Explicitly add a suppression without a diagnostic id for testing.
+            Suppression suppression = new(string.Empty) { Left = "ref/net6.0/A.dll", Right = "lib/net6.0/A.dll", IsBaselineSuppression = true };
+            string filePath = Path.Combine(Path.GetTempPath(), Path.GetTempFileName());
+
+            engine.AddSuppression(suppression);
+            Assert.True(engine.WriteSuppressionsToFile(filePath));
```

Comment:
Another interesting bit to test here with roundtripping, is if there was a suppression file already which was globally suppressed (no diagnostic id) and then we re-write the suppressions file, did  it roundtrip exactly? or did it add an empty property?